src/lib/indexer/eventProcessor.ts

```diff
@@ -9,14 +13,104 @@ const processNameSet: Set<string> = new Set(processNames)
 export const ValidateProcessName = (name: string): name is PROCESSES => processNameSet.has(name)
 
 export type EventProcessors = {
-  [key in PROCESSES]: (version: number) => ChangeSet
+  [key in PROCESSES]: (
+    version: number,
+    transaction: Transaction | null,
+    inputs: { id: number; localId: UUID }[],
+    outputs: { id: number; roles: Map<string, string>; metadata: Map<string, string> }[]
+  ) => ChangeSet
+}
+
+const getOrError = <T>(map: Map<string, T>, key: string): T => {
+  const val = map.get(key)
+  if (val === undefined) {
+    throw new Error(`Invalid token detected onchain. Missing prop ${key}`)
+  }
+  return val
 }
 
 const DefaultEventProcessors: EventProcessors = {
-  'demand-create': () => ({}),
+  'demand-create': (version, transaction, _inputs, outputs) => {
+    if (version !== 1) {
+      throw new Error(`Incompatible version ${version} for demand-create process`)
+    }
+
+    const newDemandId = outputs[0].id
+    const newDemand = outputs[0]
+
+    if (transaction) {
+      const id = transaction.localId
+      return {
+        demands: new Map([[id, { type: 'update', id, state: 'created', latest_token_id: newDemandId }]]),
```

Comment:
update `original_token_id` and `updated_at` as well?

Reply:
ok this doesn't happen automatically so I'll add it. I'll likely do it in the db module so it's always done correctly